File: haskell/zerochecker.hs
module Main where

import qualified System.Environment
import qualified Data.ByteString.Lazy.Char8 as BL

main :: IO ()
main = do
  args <- System.Environment.getArgs
  let filename = head args
  contents <- BL.readFile filename
  let result = BL.find (/= '\NUL') contents
  if result == Nothing
    then putStrLn filename
    else return ()

File: go/zerochecker.go
package main

import (
	"fmt"
	"io/ioutil"
	"os"
)

func check(e error) {
	if e != nil {
		panic(e)
	}
}

func zerofile(filename string) bool {
	data, error := ioutil.ReadFile(filename)
	check(error)
	for _, byte := range data {
		if byte != 0 {
			return false
		}
	}
	return true
}

func main() {
	filename := os.Args[1]
	if zerofile(filename) {
		fmt.Println(filename)
	}
}
